src/factories/digi/SiliconPulseDiscretization_factory.h

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2025 Simon Gardner
+
+#pragma once
+
+#include "extensions/jana/JOmniFactory.h"
+
+#include "algorithms/digi/SiliconPulseDiscretization.h"
+#include <iostream>
+
+namespace eicrecon {
+
+class SiliconPulseDiscretization_factory
+    : public JOmniFactory<SiliconPulseDiscretization_factory, SiliconPulseDiscretizationConfig> {
+public:
+  using AlgoT = eicrecon::SiliconPulseDiscretization;
+
+private:
+  std::unique_ptr<AlgoT> m_algo;
+
+  PodioInput<edm4hep::TimeSeries> m_in_pulses{this};
+  PodioOutput<edm4hep::RawTimeSeries> m_out_pulses{this};
+
+  ParameterRef<double> m_EICROC_period{this, "EICROC_period", config().EICROC_period};
+  ParameterRef<double> m_local_period{this, "local_period", config().local_period};
+  ParameterRef<double> m_global_offset{this, "global_offset", config().global_offset};
```

Comment:
CamelCase convention for JParameter names.
```suggestion
  ParameterRef<double> m_EICROC_period{this, "EICROCPeriod", config().EICROC_period};
  ParameterRef<double> m_local_period{this, "localPeriod", config().local_period};
  ParameterRef<double> m_global_offset{this, "globalOffset", config().global_offset};
```